test/unit/factorization/test_compute_t_factor.cpp

```diff
@@ -245,61 +239,55 @@ std::vector<std::tuple<SizeType, SizeType, SizeType, SizeType, SizeType, GlobalT
 //
 // Which we expect to be the equal to the one computed previously.
 template <class T, Backend B, Device D>
-void testComputeTFactor(const SizeType a_m, const SizeType a_n, const SizeType mb, const SizeType nb,
-                        const SizeType k, const GlobalTileIndex v_start) {
-  ASSERT_LE(k, nb);
+void testComputeTFactor(const SizeType m, const SizeType k, const SizeType mb, const SizeType nb,
+                        const GlobalElementIndex v_start) {
+  ASSERT_LE(v_start.row() + k, m);
+  ASSERT_LE(v_start.col() + k, nb);
 
   const TileElementSize block_size(mb, nb);
 
-  Matrix<const T, Device::CPU> v_input_h = [&, &a_m = a_m, &a_n = a_n]() {
-    Matrix<T, Device::CPU> V({a_m, a_n}, block_size);
-    dlaf::matrix::util::set_random(V);
-    return V;
-  }();
-
-  GlobalElementSize v_size(a_m, a_n);
-
-  const GlobalElementIndex v_start_el(v_start.row() * mb, v_start.col() * nb);
-  if (!v_size.isEmpty()) {
-    const GlobalElementIndex v_end_el(a_m, std::min((v_start.col() + 1) * nb, a_n));
-    v_size = v_end_el - v_start_el;
+  Matrix<T, Device::CPU> v_h({m, nb}, block_size);
+  dlaf::matrix::util::set_random(v_h);
+  auto dist_v = v_h.distribution();
+
+  // set up HHReflectors correctly.
+  for (SizeType i = v_start.row(); i < v_start.row() + k && i < m; ++i) {
+    SizeType j = i - v_start.row() + v_start.col();
+    GlobalElementIndex ij(i, j);
+    auto tile = v_h(dist_v.globalTileIndex(ij)).get();
+    auto ij_tile = dist_v.tileElementIndex(ij);
+    tile(ij_tile) = T{1};
+    for (SizeType jj = j + 1; jj < v_start.col() + k; ++jj) {
+      tile({ij_tile.row(), jj}) = T{0};
+    }
   }
 
-  MatrixLocal<T> v(v_size, block_size);
-
-  const GlobalTileSize v_offset{v_start.row(), v_start.col()};
-  for (const auto& ij_tile : iterate_range2d(v.nrTiles())) {
-    // copy only the panel
-    const auto& source_tile = v_input_h.read(ij_tile + v_offset).get();
-    matrix::internal::copy(source_tile, v.tile(ij_tile));
-    if (ij_tile.row() == 0)
-      tile::internal::laset<T>(blas::Uplo::Upper, 0.f, 1.f, v.tile(ij_tile));
-  }
+  auto v_local = dlaf::matrix::test::allGather<T>(blas::Uplo::General, v_h);
```

Comment:
If I'm not wrong, you could enforce `const`-correctness on `v_local`.

```suggestion
  auto v_local = dlaf::matrix::test::allGather<const T>(blas::Uplo::General, v_h);
```